pkg/cloudprovider/gce/gce.go

```diff
@@ -379,11 +383,40 @@ func (gce *GCECloud) CreateTCPLoadBalancer(name, region string, externalIP net.I
 		return nil, err
 	}
 
+	allowedPorts := make([]string, len(ports))
+	for ix := range ports {
+		allowedPorts[ix] = strconv.Itoa(ports[ix].Port)
+	}
+
+	hostTag := gce.computeHostTag(hosts[0])
+
+	firewall := &compute.Firewall{
+		Name:         makeFirewallName(name),
+		Description:  fmt.Sprintf("KubernetesAutoGenerated_OnlyAllowTrafficForDestinationIP_%s", fwd.IPAddress),
+		SourceRanges: []string{"0.0.0.0/0"},
+		TargetTags:   []string{hostTag},
+		Allowed: []*compute.FirewallAllowed{
+			{
+				IPProtocol: "tcp",
+				Ports:      allowedPorts,
+			},
+		},
+	}
+	if _, err = gce.service.Firewalls.Insert(gce.projectID, firewall).Do(); err != nil {
```

Comment:
What happens when someone deletes the load-balancer in GCE but not the firewall rule?  The controller will try to recreate the LB, which will come through here and fail EEXISTS - is it worth detecting that it exists and updating it instead?